atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/SweepMetrics.java

```diff
@@ -0,0 +1,107 @@
+import java.util.Arrays;
+import java.util.function.Consumer;
+
+import org.mpierce.metrics.reservoir.hdrhistogram.HdrHistogramReservoir;
+
+import com.codahale.metrics.Histogram;
+import com.codahale.metrics.Metric;
+import com.codahale.metrics.MetricRegistry;
+import com.palantir.atlasdb.keyvalue.api.SweepResults;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.util.AtlasDbMetrics;
+
+class SweepMetrics {
+    enum TableAndAggregateMetric {
+        STALE_VALUES_DELETED_METRIC(STALE_VALUES_DELETED),
+        CELLS_EXAMINED_METRIC(CELLS_EXAMINED);
+
+        private final String name;
+
+        TableAndAggregateMetric(String name) {
+            this.name = name;
+        }
+
+        void registerAggregateMetric() {
+            registerMetricWithHdrHistogram(aggregateMetric());
+        }
+
+        void registerForTable(TableReference tableRef) {
+            registerMetricWithHdrHistogram(getTableSpecificName(name, tableRef));
+        }
+
+        void recordMetric(TableReference tableRef, long value) {
+            String tableSpecificMetric = getTableSpecificName(name, tableRef);
+
+            metricRegistry.histogram(tableSpecificMetric).update(value);
+            metricRegistry.histogram(aggregateMetric()).update(value);
+        }
+
+        private String aggregateMetric() {
+            return MetricRegistry.name(SweepMetrics.class, name);
+        }
+
+        private String getTableSpecificName(String root, TableReference tableRef) {
+            return MetricRegistry.name(SweepMetrics.class, root, tableRef.getQualifiedName());
+        }
+
+        private void registerMetricWithHdrHistogram(String metric) {
+            registerMetricIfNotExists(metric, new Histogram(new HdrHistogramReservoir()));
+        }
+
+        private void registerMetricIfNotExists(String metricName, Metric metric) {
+            if (!metricRegistry.getMetrics().containsKey(metricName)) {
+                metricRegistry.register(metricName, metric);
+            }
+        }
+    }
+
+    static final String STALE_VALUES_DELETED = "staleValuesDeleted";
+    static final String CELLS_EXAMINED = "cellsExamined";
+
+    private static final MetricRegistry metricRegistry = AtlasDbMetrics.getMetricRegistry();
+
+    public static SweepMetrics create() {
+        SweepMetrics sweepMetrics = new SweepMetrics();
+        sweepMetrics.registerAggregateMetrics();
+        return sweepMetrics;
+    }
+
+    protected SweepMetrics() {
+        // To prevent instantiation
+    }
+
+    private void registerAggregateMetrics() {
+        forAllMetrics(TableAndAggregateMetric::registerAggregateMetric);
+    }
+
+    void registerMetricsIfNecessary(TableReference tableRef) {
+        forAllMetrics(metric -> metric.registerForTable(tableRef));
+    }
+
+    void recordMetrics(TableReference tableRef, SweepResults results) {
+        TableAndAggregateMetric.CELLS_EXAMINED_METRIC.recordMetric(tableRef, results.getCellsExamined());
```

Comment:
You could also have a function in the constructor of the enum that takes SweepResults and gets the appropriate result for each. Probably not worth it at this point, but should be considered when we add more metrics.